lib/index.js

```diff
@@ -259,7 +263,12 @@ exports.tooManyRequests = function (message, data) {
 
 exports.internal = function (message, data, statusCode) {
 
-    var error = (data instanceof Error ? exports.wrap(data, statusCode, message) : exports.create(statusCode || 500, message));
+    return internals.internal(message, data, statusCode, exports.internal);
+};
+
+internals.internal = function (message, data, statusCode, ctor) {
```

Comment:
Also, `internals.internal()` accepts arguments in different order than `internals.create()`. Is there a reason for that?

Since all calls to `internals.internal()` pass a status code, wouldn't be easier to just do everything in `internals.create()` by checking the status code?